File: c#_SDK/PayHubSDK/PayHubSDK/com/payhub/ws/model/AuthOnly.cs
using PayHubSDK.com.payhub.ws.model;
using PayHubSDK.com.payhub.ws.api;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Net;
using System.Runtime.Serialization;
using System.Text;
using System.Threading.Tasks;
using PayHubSDK.com.payhub.ws.util;

namespace PayHubSDK.com.payhub.ws.model
{
    ///  @author Agustin Breit 
    /// 
    [DataContract]
    public class AuthOnly : WsConnections
    {
        public static string AUTH_ID_LINK = "authonly/";
        private string _url_operation = "authonly/";
        private string _url_basePath;
        public string _url
        {
            get { return this._url_basePath + this._url_operation; }
            set { if (value != null)this._url_basePath = value; }
        }
        [DataMember]
        private Merchant merchant;
        public Merchant Merchant
        {
            get { return this.merchant; }
            set
            {
                if (value != null)
                {
                    this.merchant = value;
                }
            }
        }

        //public Merchant merchant { get; set; }
        [DataMember]
        private Customer customer;
        public Customer Customer
        {
            get { return this.customer; }
            set
            {
                if (value != null)
                {
                    this.customer = value;
                }
            }
        }
        [DataMember]
        private Bill bill;
        public Bill Bill
        {
            get { return this.bill; }
            set
            {
                if (value != null)
                {
                    this.bill = value;
                }
            }
        }
        [DataMember]
        private CardData card_data;
        
        public CardData Card_data
        {
            get { return this.card_data; }
            set
            {
                if (value != null)
                {
                    this.card_data = value;
                }
            }
        }
        public AuthOnly(Merchant merchant, Bill bill, CardData card_data,Customer customer)
        {
            this.merchant = merchant;
            this.bill = bill;
            this.card_data = card_data;
            this.customer = customer;
        }

        public AuthorizationResponseInformation authonly(string json, HttpWebRequest request)
        {
            AuthorizationResponseInformation responseObject = new AuthorizationResponseInformation();
            using (var streamWriter = new StreamWriter(request.GetRequestStream()))
            {
                streamWriter.Write(json);
                streamWriter.Flush();
                streamWriter.Close();
            }
            var result = doPost(request, _url);
            responseObject = JsonConvert.DeserializeObject<AuthorizationResponseInformation>(result);
            responseObject.rowData = result;
            return responseObject;

        }

       
    }
}

File: Payhub - SDK/java_SDK/PayHubSDK/src/main/java/com/payhub/ws/model/AuthOnly.java
/**
 * 
 */
package com.payhub.ws.model;

import java.io.BufferedReader;
import java.io.DataOutputStream;
import java.io.IOException;
import java.io.InputStreamReader;
import java.net.HttpURLConnection;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.core.JsonGenerationException;
import com.fasterxml.jackson.databind.DeserializationFeature;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.payhub.ws.api.AuthorizationResponseInformation;
import com.payhub.ws.util.WsConnections;

/**
 * @author agustin
 *
 */
@JsonInclude(JsonInclude.Include.NON_NULL) 
public class AuthOnly extends WsConnections{

	 public static String AUTH_ID_LINK = "authonly/";
	 @JsonIgnore
     private String url_basePath;
	 @JsonIgnore
     public String url;
     private Merchant merchant;
     private Customer customer;     
     private Bill bill;     
     private CardData card_data;
     
     
     public String getUrl_basePath() {
		return url_basePath;
	}

	public void setUrl_basePath(String url_basePath) {
		this.url_basePath = url_basePath;
	}

	public String getUrl() {
		return this.url_basePath+AUTH_ID_LINK;
	}

	public void setUrl(String url) {
		this.url_basePath = url;
	}

	public Merchant getMerchant() {
		return merchant;
	}

	public void setMerchant(Merchant merchant) {
		this.merchant = merchant;
	}

	public Customer getCustomer() {
		return customer;
	}

	public void setCustomer(Customer customer) {
		this.customer = customer;
	}

	public Bill getBill() {
		return bill;
	}

	public void setBill(Bill bill) {
		this.bill = bill;
	}

	public CardData getCard_data() {
		return card_data;
	}

	public void setCard_data(CardData card_data) {
		this.card_data = card_data;
	}

	public AuthOnly(Merchant merchant, Bill bill, CardData card_data,Customer customer)
     {
         this.merchant = merchant;
         this.bill = bill;
         this.card_data = card_data;
         this.customer = customer;
     }

	public AuthorizationResponseInformation authOnly(String json,HttpURLConnection request) {
		DataOutputStream wr;
		AuthorizationResponseInformation responseObject = new AuthorizationResponseInformation(); 
		try {
			wr = new DataOutputStream(request.getOutputStream());
			wr.writeBytes(json);
			wr.flush();
			wr.close();						      
	        String result = doPost(request, this.getUrl());   
	        ObjectMapper mapper = new ObjectMapper();
	        mapper.disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
	    	try {	    
	    		// read from file, convert it to user class
	    		responseObject = mapper.readValue(result, AuthorizationResponseInformation.class);
	    		responseObject.setRowData(result);
	     
	    	} catch (JsonGenerationException e) {
	     
	    		e.printStackTrace();
	     
	    	}
		} catch (IOException e1) {
			// TODO Auto-generated catch block
			try {
				BufferedReader in1 = new BufferedReader(new InputStreamReader(request.getErrorStream()));
				String inputLine1;
				StringBuffer response1 = new StringBuffer();
				while ((inputLine1 = in1.readLine()) != null) {
					response1.append(inputLine1);
				}
					in1.close();
					ObjectMapper mapper = new ObjectMapper();
			        mapper.disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
			        responseObject = mapper.readValue(response1.toString(), AuthorizationResponseInformation.class);
					responseObject.setRowData(response1.toString());
				 	
			} catch (IOException e) {
				// TODO Auto-generated catch block
				e.printStackTrace();
			}
			
		}                    
        return responseObject;
	}

}
